clients/ethereum/client.go

```diff
@@ -46,6 +48,7 @@ type ArgsEthereumClient struct {
 }
 
 type client struct {
+	ethclient                 *ethclient.Client
```

Comment:
please remove this. Instead, extend the ClientWrapper interface with the BalanceAt method.